sw/airborne/modules/datalink/mavlink.c

```diff
@@ -21,553 +21,353 @@
-#else
-// use radio_control.values for now...
-#define RC_CHANNELS RADIO_CONTROL_NB_CHANNEL
-#define PPM_OF_PPRZ(_v) ((_v) / 19.2 + 1500)
-#define PPM_PULSES(_i) ((_i) < RADIO_CONTROL_NB_CHANNEL ? PPM_OF_PPRZ(radio_control.values[(_i)]) + MAX_PPRZ : UINT16_MAX)
-#endif
-#endif
+  // Initialize the host address struct
+  memset(&rem_addr, 0, sizeof(rem_addr));
+  rem_addr.sin_family = AF_INET; // Internet Protocol
+  rem_addr.sin_addr.s_addr = inet_addr(MAVLINK_IP_ADDRESS_STR);
+  // rem_addr.sin_addr.s_addr = inet_addr("192.168.42.3");
+  rem_addr.sin_port = htons(MAVLINK_UDP_PORT);
 
-static inline void mavlink_send_rc_channels(void)
+  // Send the current block
+  mavlink_block_cb(mission_mgr.current_block);
+}
+
+/**
+ * Periodic MAVLink calls.
+ * Called at MAVLINK_PERIODIC_FREQUENCY (10Hz)
+ */
+void mavlink_periodic(void)
 {
-#if defined RADIO_CONTROL
-  mavlink_msg_rc_channels_send(MAVLINK_COMM_0,
-                               get_sys_time_msec(),
-                               RC_CHANNELS,
-                               PPM_PULSES(0), PPM_PULSES(1), PPM_PULSES(2),
-                               PPM_PULSES(3), PPM_PULSES(4), PPM_PULSES(5),
-                               PPM_PULSES(6), PPM_PULSES(7), PPM_PULSES(8),
-                               PPM_PULSES(9), PPM_PULSES(10), PPM_PULSES(11),
-                               PPM_PULSES(12), PPM_PULSES(13), PPM_PULSES(14),
-                               PPM_PULSES(15), PPM_PULSES(16), PPM_PULSES(17),
-                               255); // rssi unknown
-#else
-  mavlink_msg_rc_channels_send(MAVLINK_COMM_0,
-                               get_sys_time_msec(),
-                               0, // zero channels available
-                               UINT16_MAX, UINT16_MAX, UINT16_MAX, UINT16_MAX,
-                               UINT16_MAX, UINT16_MAX, UINT16_MAX, UINT16_MAX,
-                               UINT16_MAX, UINT16_MAX, UINT16_MAX, UINT16_MAX,
-                               UINT16_MAX, UINT16_MAX, UINT16_MAX, UINT16_MAX,
-                               UINT16_MAX, UINT16_MAX, 255);
-#endif
+  RunOnceEvery(10, mavlink_send_heartbeat()); // Call at 1Hz
+  RunOnceEvery(1, mavlink_send_attitude()); // Call at 10Hz
+  RunOnceEvery(1, mavlink_send_altitude_ground_speed()); // Call at 10Hz
+  RunOnceEvery(10, mavlink_send_battery_status()); // Call at 1Hz
+  RunOnceEvery(10, mavlink_send_gps_status()); // Call at 1Hz
+  RunOnceEvery(1, mavlink_send_gps_position()); // Call at 10Hz
+
+  // Update the current block index in the block manager (does not need to run as fast as possible)
+  if (mission_mgr.current_block != nav_block) {
+    mission_mgr.current_block = nav_block;
+    mavlink_block_cb(mission_mgr.current_block);
+  }
 }
 
-#include "subsystems/electrical.h"
-static inline void mavlink_send_battery_status(void)
+/**
+ * Event loop MAVLink calls
+ * Called at the end of the event loop (runs at maximum allowable frequency)
+ */
+void mavlink_event(void)
```

Comment:
What's wrong with the previous way the actual connection (UDP, UART, etc.) was handled? It was working for bare metal and linux systems and uart and not only linux & UDP